includes/payment-methods/class-wc-stripe-upe-payment-method-multibanco.php

```diff
@@ -29,6 +29,104 @@ public function __construct() {
 		);
 
 		add_filter( 'wc_stripe_allowed_payment_processing_statuses', [ $this, 'add_allowed_payment_processing_statuses' ], 10, 2 );
+
+		add_action( 'wc_gateway_stripe_process_payment_intent_requires_action', [ $this, 'save_instructions' ], 10, 2 );
+		add_action( 'woocommerce_thankyou_stripe_multibanco', [ $this, 'thankyou_page' ] );
+		add_action( 'woocommerce_email_before_order_table', [ $this, 'email_instructions' ], 10, 3 );
+	}
+
+	/**
+	 * Output for the order received page.
+	 *
+	 * @param int $order_id
+	 */
+	public function thankyou_page( $order_id ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order ) {
+			return;
+		}
+
+		$this->get_instructions( $order );
+	}
+
+	/**
+	 * Add content to the WC emails.
+	 *
+	 * @param WC_Order $order
+	 * @param bool     $sent_to_admin
+	 * @param bool     $plain_text
+	 */
+	public function email_instructions( $order, $sent_to_admin, $plain_text = false ) {
+		$payment_method = $order->get_payment_method();
+		if ( ! $sent_to_admin && 'stripe_multibanco' === $payment_method && $order->has_status( 'on-hold' ) ) {
+			$this->get_instructions( $order, $plain_text );
+		}
+	}
+
+	/**
+	 * Gets Multibanco payment instructions for the customer.
+	 *
+	 * @param WC_Order $order
+	 */
+	public function get_instructions( $order, $plain_text = false ) {
+		$data = $order->get_meta( '_stripe_multibanco' );
+		if ( ! $data ) {
+			return;
+		}
+
+		if ( $plain_text ) {
+			esc_html_e( 'MULTIBANCO ORDER INFORMATION', 'woocommerce-gateway-stripe' ) . "\n\n";
+			echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n\n";
+			esc_html_e( 'Amount:', 'woocommerce-gateway-stripe' ) . "\n\n";
+			echo wp_kses_post( $data['amount'] ) . "\n\n";
+			echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n\n";
+			esc_html_e( 'Entity:', 'woocommerce-gateway-stripe' ) . "\n\n";
+			echo esc_html( $data['entity'] ) . "\n\n";
+			echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n\n";
+			esc_html_e( 'Reference:', 'woocommerce-gateway-stripe' ) . "\n\n";
+			echo esc_html( $data['reference'] ) . "\n\n";
+			esc_html_e( 'Voucher URL:', 'woocommerce-gateway-stripe' ) . "\n\n";
+			echo esc_html( $data['hosted_voucher_url'] ) . "\n\n";
```

Comment:
In the `save_instructions` method we are saving only amount, entity and reference. `hosted_voucher_url` is missing there.

Reply:
Good catch. I was initially storing and displaying that field, but decided against it in the end. Removed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3664/commits/063a5b5ad78c28963d80c8b44e05e162840d3fcb